src/contexts/DeployContext.tsx

```diff
@@ -0,0 +1,73 @@
+import { toast } from 'react-toastify'
+import { useTdpClient } from 'src/contexts'
+
+import { DeployRequest, FilterTypeEnum } from '@/client-sdk'
+import { useReducer } from 'react'
+import { deployReducer } from 'src/utils/deployReducer'
+
+import { createContext, useContext } from 'react'
+import { DeployActions } from 'src/utils/deployReducer'
+import { DeployStateType } from '../types/deployTypes'
+
+const initialState = {
+  operations: [],
+  filterExpression: '',
+  filterType: FilterTypeEnum.Regex,
+  selectedDeployMode: null,
+  restart: false,
+}
+
+const DeployContext = createContext<{
+  deploy: () => Promise<void>
+  state: DeployStateType
+  dispatch: React.Dispatch<DeployActions>
+}>(null)
+
+export function DeployContextProvider({ children }) {
+  const { deployApi } = useTdpClient()
+  const [state, dispatch] = useReducer(deployReducer, initialState)
+
+  async function deploy() {
+    if (!state.selectedDeployMode) {
+      toast.error('Please select a deploy mode')
+      return
+    }
+    if (state.selectedDeployMode === 'custom') {
+      if (!state.operations.length) {
+        toast.error('Please select at least one operation')
+        return
+      }
+      const res = await deployApi.runNodesApiV1DeployRunPost({
+        targets: state.operations,
+      })
+      res?.data?.state && toast.info(`Deploy status: ${res.data.state}`)
+      return
+    }
+    const deployReq: DeployRequest = { restart: state.restart }
+    if (state.filterExpression.trim()) {
+      deployReq.filter_type = state.filterType
+      deployReq.filter_expression = state.filterExpression
+    }
+    if (state.selectedDeployMode !== 'all') {
+      deployReq[state.selectedDeployMode] = state.operations
+    }
+    const res = await deployApi.deployNodeApiV1DeployPost(deployReq)
+    res?.data?.state && toast.info(`Deploy status: ${res.data.id}`)
```

Comment:
```suggestion
    res?.data?.state && toast.info(`Deployment: ${res.data.id}`)
```

Status does not make sense